db/db_impl/db_impl.cc

```diff
@@ -4417,6 +4419,19 @@ Status DBImpl::IngestExternalFiles(
         versions_->SetLastPublishedSequence(last_seqno + consumed_seqno_count);
         versions_->SetLastSequence(last_seqno + consumed_seqno_count);
       }
+    }
+
+    // The following LogAndApply() can cause very high latency up to seconds. If
```

Comment:
Can you explain why LogAndApply() can cause high latency? Are we doing more than fsync()?

Reply:
Also note - the `FlushMemTable()` on line 4385 blocks writes while it runs, and we often see a multi-second latency come from there.

Reply:
@siying The "manifest.file.sync.micros" from RocksDB shows that it can take one or two seconds sometimes.

This is how the statistics look like across hundreds of instances.
![屏幕快照 2020-09-09 10 12 30](https://user-images.githubusercontent.com/7806468/92546414-0bc12180-f285-11ea-8e21-38c9aa98df43.png)

Reply:
@ajkr Yes, that's a problem too. But I have figured out a way to work around that in our scenario. https://github.com/tikv/rust-rocksdb/blob/master/librocksdb_sys/crocksdb/c.cc#L4241

Reply:
@huachaohuang I trust what you see. What I didn't understand is why it takes so long. Is it that one fsync() takes more than one second, or we are doing more than one for this bulkloading path? If it is the latter case, maybe we should fix the root cause.

Reply:
@siying I tried to add more statistics around `SyncManifest()` and it did take quite long sometimes. And I think "manifest.file.sync.micros" also showed similar results. Maybe a better disk can reduce the latency, but that's not up to me :(

Reply:
@huachaohuang is it reproducible? If it is reproducible, maybe strace fsync and match the bulkload calls to fsync calls issued for the command. We will discuss internally about this feature request in the mean time.